File: Interview Preparation/Trees/huffman.java

class Decoding {

    /*
     * class Node public int frequency; // the frequency of this tree public char
     * data; public Node left, right;
     * 
     */

    void decode(String s, Node root) {
        Node n = root;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '0') {
                n = n.left;
            } else {
                n = n.right;
            }
            if (n.left == null && n.right == null) {
                System.out.print(n.data);
                n = root;
            }
        }
    }

}

File: Interview Preparation/Trees/huffman.py
"""class Node:
    def __init__(self, freq,data):
        self.freq= freq
        self.data=data
        self.left = None
        self.right = None
"""        

# Enter your code here. Read input from STDIN. Print output to STDOUT
def decodeHuff(root, s):
	#Enter Your Code Here
    n = root
    string = ''

    for i in range(len(s)):
        if s[i] == '0':
            n = n.left
        else:
            n = n.right
        
        if n.left == None and n.right == None:
            string = string + n.data
            n = root
            
    print(string)